src/main/java/Deadline.java

```diff
@@ -0,0 +1,55 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task that has a deadline which a user is to meet.
+ */
+public class Deadline extends Task {
+
+    protected String by;
+
+    /**
+     * Construct a task object that has a deadline to meet.
+     * @param description Description of task
+     * @param by Date and time to be due
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String getDate() {
+        return this.by;
+    }
+
+    /**
+     * Returns a date and time object of the specific date.
+     *
+     * @return Date and time Object.
+     */
+    public LocalDateTime getLocalDateTime() {
+        String dateStr = this.getDate();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```

Comment:
I notice that this code is repeated in Event class as well. Perhaps you can abstract out the common attributes and methods in Event and Deadline classes using a super class and make Event and Deadline extend from that super class.